components/aws/sagemaker/common/_utils.py

```diff
@@ -471,11 +471,61 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  endpoint_exists = False
+  if endpoint_name:
+      endpoints = client.list_endpoints(
+          MaxResults=99,
+          NameContains=endpoint_name,
+      )['Endpoints']
+      for i in range(len(endpoints)):
+          ep = endpoints[i]
+          if (ep['EndpointName'] == endpoint_name):
+            endpoint_exists = True
+            logging.info("Endpoint exists: " + ep['EndpointName'] + " - " + ep['EndpointArn'])
+  return endpoint_exists
+
+def endpoint_config_name_exists(client, endpoint_config_name):
+  ep_config_name_exists = False
+  if endpoint_config_name:
+      if client.describe_endpoint_config(
+          EndpointConfigName=endpoint_config_name,
+          )['EndpointConfigName'] == endpoint_config_name:
+        ep_config_name_exists = True
+        logging.info("Endpoint Config exists: " + endpoint_config_name)
+  return ep_config_name_exists
+
 def deploy_model(client, args):
-  endpoint_config_name = create_endpoint_config(client, args)
-  endpoint_name = create_endpoint(client, args['region'], args['endpoint_name'], endpoint_config_name, args['endpoint_tags'])
+  if args['update_endpoint'] and endpoint_name_exists(client, args['endpoint_name']):
+      endpoint_config_name = create_endpoint_config(client, args)
+      endpoint_name = update_endpoint(client, args['region'], args['endpoint_name'], endpoint_config_name)
+  else:
+      args['update_endpoint'] = False
+      endpoint_config_name = create_endpoint_config(client, args)
+      endpoint_name = create_endpoint(client, args['region'], args['endpoint_name'], endpoint_config_name, args['endpoint_tags'])
   return endpoint_name
 
+def get_current_endpoint_config(client, args):
+    endpoint_config_name = ''
+    try:
+        endpoint_config_name = client.describe_endpoint(EndpointName=args['endpoint_name'])['EndpointConfigName']
+        logging.info("Current Endpoint Config Name: " + endpoint_config_name)
+    except ClientError as e:
+        logging.info("Endpoint Config does not exist")
+        ## This is not an error, end point may not exist
+    return endpoint_config_name
+
+def cleanup_endpoint_config(client, endpoint_config_name, args):
+    try:
+        if args['update_endpoint'] and endpoint_config_name:
+            current_endpoint_config = get_current_endpoint_config(client, args)
+            logging.info('Starting best effort endpoint config cleanup for the update...')
```

Comment:
can this message could be confusing to customers?

Reply:
Changed the logging message to say whether the endpoint is deleted or not.